services/worker-manager/src/providers/aws.js

```diff
@@ -254,6 +254,39 @@ class AwsProvider extends Provider {
     }));
   }
 
+  async removeWorker({worker}) {
+    let result;
+    try {
+      result = await this.ec2s[worker.providerData.region].terminateInstances({
+        InstanceIds: [worker.workerId],
+      }).promise();
+    } catch (e) {
+      this.monitor.err(`Error terminating AWS instance: ${e}`);
+
+      const workerPool = this.WorkerPool.load({
+        workerPoolId: worker.workerPoolId,
+      });
+      await workerPool.reportError({
+        kind: 'termination-error',
+        title: 'Instance Termination Error',
+        description: e.message,
+        notify: this.notify,
+        WorkerPoolError: this.WorkerPoolError,
+      });
+
+      throw e;
+    }
+
+    result.TerminatingInstances.forEach(ti => {
+      if (!ti.InstanceId === worker.workerId || !ti.CurrentState.Name === 'shutting-down') {
```

Comment:
When would the first half of this be untrue? If it can be false, let's have a different error since this error won't actually say what happened. Will either of these ever be true actually?

Reply:
I'd say it's highly unlikely.... In case of a bug I guess

> Will either of these ever be true actually?
I don't know. Seemed like a sensical check

Reply:
I changed the error message to be one combined "Unknown error"